src/Microsoft.Extensions.Http.Polly/PolicyExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+
+namespace Polly
+{
+    public static class PolicyExtensions
+    {
+        public static PolicyBuilder<HttpResponseMessage> HandleTransientHttpError()
+        {
+            return Policy.Handle<HttpRequestException>().OrTransientHttpStatusCode();
+        }
+
+        public static PolicyBuilder<HttpResponseMessage> OrTransientHttpStatusCode(this PolicyBuilder policyBuilder)
+        {
+            if (policyBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(policyBuilder));
+            }
+
+            return policyBuilder.OrResult<HttpResponseMessage>(response =>
+            {
+                return (int)response.StatusCode >= 500 || response.StatusCode == HttpStatusCode.RequestTimeout;
```

Comment:
Could `429` be considered "transient", or is that something that would be handled by something more specialised?

Reply:
Good question, I think for now we're intentionally leaving it out. You'd want to handle that differently than a timeout or a transient connection issue.

Reply:
Agree about probably not including `429` here:

+ We can expect users to use `OrTransientHttpStatusCode()` with policies other than Retry, for example CircuitBreaker.  We may not want to embed _as a default_ that `429` is bad enough news to contribute towards breaking a circuit. 

+ As you say @martincostello and @rynowak , 429 can play nicely with specialised handling.  For example Polly has [WaitAndRetry(...) overloads](https://github.com/App-vNext/Polly/wiki/Retry#retryafter-when-the-response-specifies-how-long-to-wait) enabling the policy to wait before retrying for exactly the duration specified in any `Retry-After` or `x-ms-retry-after-ms` header.  If we embed `429` in `OrTransientHttpStatusCode()`, we force users to handle 429 and other retry-able causes in the same policy, possibly leading to internal forking logic and limiting options for treating them differently.   

`429` could be covered by some documentation examples.